src/components/Custom/RJSFFormFieldAdapter/CustomPriorityBoundsField.jsx

```diff
@@ -169,12 +172,140 @@ const CustomPriorityBoundsField = (props) => {
     }
   }, [isMapVisible]);
 
+  // Close tooltip when clicking outside
+  useEffect(() => {
+    const handleClickOutside = (event) => {
+      if (showTooltip && !event.target.closest(".tooltip-container")) {
+        setShowTooltip(false);
+      }
+    };
+
+    document.addEventListener("click", handleClickOutside);
+    return () => document.removeEventListener("click", handleClickOutside);
+  }, [showTooltip]);
+
   const handleChange = (newData) => {
     if (typeof props.onChange === "function") {
       props.onChange(Array.isArray(newData) ? [...newData] : newData);
     }
   };
 
+  // Validate GeoJSON structure
+  const validateGeoJSON = (geoJson) => {
+    if (!geoJson || typeof geoJson !== "object") {
+      return { valid: false, error: "Invalid JSON format" };
+    }
+
+    if (!Array.isArray(geoJson.features)) {
+      return { valid: false, error: "Features must be an array" };
+    }
+
+    const polygonFeatures = geoJson.features.filter(
+      (feature) =>
+        feature.type === "Feature" && feature.geometry && feature.geometry.type === "Polygon",
+    );
+
+    if (polygonFeatures.length === 0) {
+      return { valid: false, error: "No valid Polygon features found" };
+    }
+
+    return { valid: true, features: polygonFeatures };
+  };
+
+  // Handle file upload
+  const handleFileUpload = async (file) => {
+    setUploadFeedback(null);
+
+    // Check file type
+    const validExtensions = [".json", ".geojson"];
+    const fileExtension = file.name.toLowerCase().substring(file.name.lastIndexOf("."));
+
+    if (!validExtensions.includes(fileExtension)) {
+      setUploadFeedback({ type: "error", message: messages.fileTypeError });
+      return;
+    }
+
+    try {
+      const text = await file.text();
+      const geoJson = JSON.parse(text);
+
+      const validation = validateGeoJSON(geoJson);
+      if (!validation.valid) {
+        setUploadFeedback({
+          type: "error",
+          message: messages.invalidGeoJSON,
+          details: validation.error,
+        });
+        return;
+      }
```

Comment:
The function signature of `validateGeoJSON` is odd. A more typical pattern would be to either:
- return a boolean (true if valid, false otherwise), if detailed error messages are not necessary
- return nothing (`undefined`) on success, or throw an `Error` with an appropriate message if malformed.

Either of these would make the call site code simpler. The current code uses a pattern similar to Alexis King's [Parse, don't validate](https://lexi-lambda.github.io/blog/2019/11/05/parse-don-t-validate/), which is good advice in strongly typed languages, but in plain JS where we don't have a type system to distinguish the pre-validation and post-validation objects at the type level, only serves to confuse things.